pkg/instancetype/instancetype.go

```diff
@@ -40,7 +40,7 @@ const (
 
 type Methods interface {
 	FindInstancetypeSpec(vm *virtv1.VirtualMachine) (*instancetypev1beta1.VirtualMachineInstancetypeSpec, error)
-	ApplyToVmi(field *k8sfield.Path, instancetypespec *instancetypev1beta1.VirtualMachineInstancetypeSpec, prefernceSpec *instancetypev1beta1.VirtualMachinePreferenceSpec, vmiSpec *virtv1.VirtualMachineInstanceSpec) Conflicts
+	Apply(field *k8sfield.Path, instancetypespec *instancetypev1beta1.VirtualMachineInstancetypeSpec, preferenceSpec *instancetypev1beta1.VirtualMachinePreferenceSpec, vmiSpec *virtv1.VirtualMachineInstanceSpec, vmiMetadata *metav1.ObjectMeta) Conflicts
```

Comment:
Why rename `ApplyToVmi` to `Apply` here? Unless `ApplyToVmi` is no longer descriptive of the function's behavior isn't it better to leave it as is?

Good catch on the type of `preferenceSpec` though.

Reply:
I have dropped it for now.